### DISCUSSION TITLE
next/image Loader (placeholder)

### DISCUSSION THREAD

**Discussion by 1jmj** [vercel/next.js#25899] [Ideas]
### Describe the feature you'd like to request

While an image is being loaded by Next/Image, sometimes there is a delay and empty space is shown. It would be good to show a lqip, loader, spinner, or other component like a Placeholder in place of the image until it is loaded.

### Describe the solution you'd like

Props to allow a component to be displayed until next/image completes the load of it's image.

### Describe alternatives you've considered

I have tried using other methods of placeholder which are unloaded when main image is loaded but they are not clean as next/image does not report that it has completed loading.

**Comment by jacobhq** [comment]
This has been added in nextjs 11, see [these docs](https://nextjs.org/docs/api-reference/next/image#placeholder). v11 was released after you opened this discussion.

**Reply by 1jmj** [reply]
Indeed. I saw the event this week and was gutted that I implemented the same functionality just before the released it!!! Oh well, learned loads!

**Reply by jacobhq** [reply]
Have a great day

**Reply by VityaSchel** [reply]
I can't get placeholder working :(
I load my image dynamically, my server sends original image url with base64-encoded data URL. I set them at the same time with hooks inside Image component like this:

```javascript
setPictureURL("https://ik.imagekit.io/splooit/forgithubissue.png")
setPlaceholderDataURL("data:image/png;base64,/9j/2wBDAAEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQH/2wBDAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQH/wAARCAAKAAoDAREAAhEBAxEB/8QAFwAAAwEAAAAAAAAAAAAAAAAAAAIJCv/EACMQAAAFAwMFAAAAAAAAAAAAAAEEBQYHAggRAAMJEiFBUaH/xAAUAQEAAAAAAAAAAAAAAAAAAAAA/8QAFBEBAAAAAAAAAAAAAAAAAAAAAP/aAAwDAQACEQMRAD8AzH2uWrcZku8ec2SZOt7RW3q9llOx/LcUxiok1lzlpPaTHYCcupTABok2qWTkg/JS6bNIrWkMxI9Fae4Sm4mHGSdT9mo4ZCOQBRjvWID5Dpzj7oEyPsdAaD//2Q==")

<Image
  src={pictureURL}
  placeholder='blur'
  blurDataURL={placeholderDataURL}
/>
```

But it seems like Image component just doesn't see the value in `blurDataURL` prop!!! 

It changes from previous picture to new as soon as new loaded ignoring placeholder. I also tried emptying `src` prop and set it to new after 1ms but it doesn't work neither. 

Should I open an issue here? I'm stuck with this problem please help!

**Reply by VityaSchel** [reply]
> I can't get placeholder working :(
> I load my image dynamically, my server sends original image url with base64-encoded data URL. I set them at the same time with hooks inside Image component like this:
> 
> ```js
> setPictureURL("https://ik.imagekit.io/splooit/forgithubissue.png")
> setPlaceholderDataURL("data:image/png;base64,/9j/2wBDAAEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQH/2wBDAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQEBAQH/wAARCAAKAAoDAREAAhEBAxEB/8QAFwAAAwEAAAAAAAAAAAAAAAAAAAIJCv/EACMQAAAFAwMFAAAAAAAAAAAAAAEEBQYHAggRAAMJEiFBUaH/xAAUAQEAAAAAAAAAAAAAAAAAAAAA/8QAFBEBAAAAAAAAAAAAAAAAAAAAAP/aAAwDAQACEQMRAD8AzH2uWrcZku8ec2SZOt7RW3q9llOx/LcUxiok1lzlpPaTHYCcupTABok2qWTkg/JS6bNIrWkMxI9Fae4Sm4mHGSdT9mo4ZCOQBRjvWID5Dpzj7oEyPsdAaD//2Q==")
> 
> <Image
>   src={pictureURL}
>   placeholder='blur'
>   blurDataURL={placeholderDataURL}
> />
> ```
> 
> But it seems like Image component just doesn't see the value in `blurDataURL` prop!!!
> 
> It changes from previous picture to new as soon as new loaded ignoring placeholder. I also tried emptying `src` prop and set it to new after 1ms but it doesn't work neither.
> 
> Should I open an issue here? I'm stuck with this problem please help!

For now I will use workaround: create img html tag underneath Image component using z-index and assign blurred src to it while Image component is blank and loading
